portal/views/auth.py

```diff
@@ -276,6 +276,15 @@ def picture_url(result):
         from werkzeug.datastructures import ImmutableMultiDict
         request.args = ImmutableMultiDict()
 
+    prv = 'FB' if (provider_name == 'facebook') else provider_name.upper()
+
+    if (not current_app.config.get('{}_CONSUMER_KEY'.format(prv)) or
```

Comment:
Would this also catch when `*_CONSUMER_KEY` is not defined at all (as opposed to just being an empty string)?

 I think you may need to add a second/default arg to `get()`...